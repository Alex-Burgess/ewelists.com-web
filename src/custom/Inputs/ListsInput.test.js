import * as React from 'react';
import renderer from 'react-test-renderer';
import ListsInput from './ListsInput';

test('Renders ListsInput component with title prop.', () => {
  const tree = renderer
    .create(
      <ListsInput
        id="title"
        title
        inputProps={{
          placeholder: "Enter your title here...",
          defaultValue: "Default Title",
        }}
        formControlProps={{
          fullWidth: true
        }}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('Renders ListsInput component with description prop.', () => {
  const tree = renderer
    .create(
      <ListsInput
        id="description"
        description
        formControlProps={{
          fullWidth: true
        }}
        inputProps={{
          placeholder: "Add your description here...",
          defaultValue: "Default description value",
          multiline: true,
          rows: 2
        }}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
