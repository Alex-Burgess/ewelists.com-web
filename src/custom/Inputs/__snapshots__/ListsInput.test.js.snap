// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders ListsInput component with description prop. 1`] = `
<div
  className="MuiFormControl-root MuiFormControl-fullWidth"
>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-underline-2 MuiInputBase-formControl MuiInput-formControl MuiInputBase-multiline MuiInput-multiline"
    onClick={[Function]}
  >
    <textarea
      aria-invalid={false}
      className="MuiInputBase-input MuiInput-input makeStyles-input-11 makeStyles-description-15 MuiInputBase-inputMultiline MuiInput-inputMultiline"
      defaultValue="Default description value"
      disabled={false}
      id="description"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Add your description here..."
      required={false}
      rows={2}
    />
  </div>
</div>
`;

exports[`Renders ListsInput component with title prop. 1`] = `
<div
  className="MuiFormControl-root MuiFormControl-fullWidth"
>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-underline-2 MuiInputBase-formControl MuiInput-formControl"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      className="MuiInputBase-input MuiInput-input makeStyles-input-11 makeStyles-title-13"
      defaultValue="Default Title"
      disabled={false}
      id="title"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Enter your title here..."
      required={false}
      type="text"
    />
  </div>
</div>
`;
